### PR TITLE
Stream support

### PR DESCRIPTION
**Pull request by guitarneck**
Please find : 
- Stream support is now available [gulp.src(...,{buffer:false}).pipe(...)]
- Tests upgraded and reinforced
- Minor texts corrections

:)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ node_modules
 dist
 .gulp
 gulpfile.js
+*.xcworkspace/*
```
Comment (moshisora) [inline .gitignore:6]: It seems to be for cocoapods, why is it necessary?

File: test/filter_test.js [hunk 2]
```diff
@@ -5,27 +5,59 @@ var fs = require('fs'),
...
+        capture.on();
         gulp.src('test/src/*.js')
             .pipe(diff({
                 clear: true,
                 dest: 'test/src/2.js'
             }))
             .pipe(concatStream(function (buf) {
+                let out = capture.get(true);
+                capture.off(false);
+                assert(out.indexOf('[diff log] flushing hash...') > -1);
```
Comment (moshisora) [inline test/filter_test.js]: It might be better to define a constant ?

**Comment by moshisora** [comment]
Thank you! I like this changes 👍 

I notice a point, please check it.
https://github.com/moshisora/gulp-diff-build/pull/8#discussion_r165964027

File: .gitignore (hunk 1 above)
Comment (guitarneck) [reply .gitignore:6]: I use XCode as my IDE.
It's for not polluated the repo with unnecessary directories and files.

File: test/filter_test.js (hunk 2 above)
Comment (guitarneck) [reply test/filter_test.js]: You're right. I was thinking about it also.
I'll change it today.

**Comment by guitarneck** [comment]
Glad you like it :)

I updated the tests files with constants strings for the log messages.
Reducing the test by the way. (See : method .has(s) )

**Comment by guitarneck** [comment]
Sorry, bad button for my last comment...

**Comment by moshisora** [comment]
thank you for your contribution, it'll be released as `v1.0.2` on npm

**Comment by guitarneck** [comment]
You're welcome !
It's a realy usefull plugin.

**Pull request merged**